docs/APP-SETUP.md

```diff
@@ -0,0 +1,216 @@
+
+![Example of Repo Scope](../assets/repo-scope-example.png)
+
+## GitHub App
+
+### Procedure
+
+1. Create a GitHub Organization
+2. Follow [Registering a GitHub App](https://docs.github.com/en/apps/creating-github-apps/registering-a-github-app/registering-a-github-app) to create the base of your GitHub App. You can use the default settings unless instructed otherwise.
+   1. For `Homepage URL`, `Webhook URL` and `Callback URL` you can put any value as a placeholder as we will deal with these at a later step.
+   2. Deselect `Expire user authorization tokens`.
+   3. You will need to create a `webhook secret` as part of the App setup process.
+3. You should reference [App Permissions](#app-permissions) to set the proper permissions for your GitHub App. This step is crucial in ensuring functionality.
+4. After app creation, at the top of the page you will be prompted to create a `private key`. Do this. You also need to generate a `client secret`.
+5. In the `Developer Settings` of your newly created GitHub App, navigate to `Install App` and install it to your Organization.
+
+**Note:** It is helpful as part of this step to store the following information somewhere safe:
+- GitHub App ID
+- Client ID
+- Client secret
+- Webhook secret
+- Private key
+
+### App Permissions
+
+Helpful reading can be found [here](https://backstage.io/docs/integrations/github/github-apps/#app-permissions), however, all required permissions and subscriptions have been compiled below:
+
+#### Repository Permissions
+- **Actions** - Read & Write
+- **Administration** - Read & Write
+- **Checks** - Read & Write
+- **Commit Statuses** - Read
+- **Contents** - Read & Write
+- **Environments** - Read & Write
+- **Issues** - Read & Write
+- **Metadata** - Read
+- **Pull Requests** - Read & Write
+- **Secrets** - Read & Write
+- **Variables** - Read & Write
+- **Workflows** - Read & Write
+
+#### Organization Permissions
+- **Members** - Read
+- **Plan** - Read
+
+#### Subscribed Events
+- Check suite
+- Check run
+- Commit comment
+- Issue comment
+- Pull request
+- Push
+- Repository dispatch
+- Workflow dispatch
+- Workflow job
+- Workflow run
+
+## GitLab Personal Access Token (PAT)
+
+### Prerequisites
+
+- A GitLab account.
+
+### Procedure
+
+- See [create a personal access token](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token).
+- Set the following token permissions:
+  - `api`
+  - `read_repository`
+  - `write_repository`
+
+**Note:** Remember to save your PAT somewhere safe.
+
+## GitLab App
+
+### Procedure
+
+1. Create a GitLab App for your GitLab Group.
+   1. See [create a group owned application](https://docs.gitlab.com/ee/integration/oauth_provider.html#create-a-group-owned-application).
+   2. The `Homepage / Callback / Webhook URLs` can be any value as a placeholder for now.
+
+**Note:** Store your `Application ID` and `Secret` somewhere safe.
+
+## Quay Setup
```

Comment:
I'm sure the quay token generation is required for the installer. but does it related to github/gitlab app setup?

in the following section, you mentioned store the token and docker configuration. but not mentioning where to use those token. I'm presuming it's should be included in installer configuration?

Reply:
I believe I included the Quay setup as the items are used as part of the installer https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/README.md#setting-environment-variables-for-configuration-scripts

So as part of the pre-req setup of github/gitlab app and quay you'd have all of the necessary tokens etc on hand for the installer was what I envisioned in my head, wdyt

Reply:
can you link the doc in this PR and https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/README.md#setting-environment-variables-for-configuration-scripts together?
perhaps reference the doc in the quay setup: `the token and dockerjson config will be used for installer configuration,  see https://github.com/redhat-ai-dev/ai-rhdh-installer/blob/main/README.md#setting-environment-variables-for-configuration-scripts `

and also in the installer configuration doc, you might want to reference back where you get those tokens

Reply:
Referenced here https://github.com/redhat-ai-dev/ai-rhdh-installer/pull/11/commits/b9816b1aa80f4207e9ad4a486d4a70b0626dd13a